src/System.Private.CoreLib/shared/System/Reflection/Assembly.cs

```diff
@@ -145,17 +146,21 @@ public override string ToString()
         public virtual long HostContext { get { throw NotImplemented.ByDesign; } }
 
         public override bool Equals(object o) => base.Equals(o);
+        private bool Equals(Assembly obj) => ReferenceEquals(this, obj) ? true : base.Equals(obj);
```

Comment:
Do these tiny private Equals methods make any difference? I expect that they will be pretty much always inlined and you can just fold them into the `operator==`.

Reply:
It would become a virtual call without it as can't call `base.Equals` in a static method?

Reply:
Though not sure it needs the ReferenceEquals fast path either as `base.Equals` goes straight through to object.Equals which is the same?
```
[below ALWAYS_INLINE size] MethodInfo:op_Inequality(ref,ref):bool
  [aggressive inline attribute] MethodInfo:op_Equality(ref,ref):bool
    [below ALWAYS_INLINE size] MethodInfo:Equals(ref):bool:this
      [below ALWAYS_INLINE size] MethodBase:Equals(ref):bool:this
        [below ALWAYS_INLINE size] MemberInfo:Equals(ref):bool:this
          [below ALWAYS_INLINE size] Object:Equals(ref):bool:this
```

Reply:
It has to be virtual call to the overriden Equals. Calling the `base.Equals` non-virtually is a bug.

Reply:
Ah, true

Reply:
Without extra method
```
Total bytes of diff: -1307 (-0.04% of base)
    diff is an improvement.
```
With extra method
```
Total bytes of diff: -1547 (-0.04% of base)
    diff is an improvement.
```

Reply:
e.g.
```asm
G_M16770_IG07:
       cmp      rax, rdx
       je       SHORT G_M16770_IG08
       mov      rcx, rax
       mov      rax, qword ptr [rax]
       mov      rax, qword ptr [rax+64]
       call     qword ptr [rax+16]Object:Equals(ref):bool:this
       jmp      SHORT G_M16770_IG09
```
vs
```asm
G_M16770_IG07:
       cmp      rcx, rdx
       je       SHORT G_M16770_IG08
       call     Module:Equals(ref):bool:this
       jmp      SHORT G_M16770_IG09
```